### PR TITLE
Cleanup pure pursuit and add support for enforced driving directions

### PR DESCRIPTION
**Pull request by shadaj**

### CODE REVIEW

**Comment by shadaj** [comment]
@Paxelord @RinHa ready for review!

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/unicycle/control/purePursuit/MathUtilities.scala [hunk 1]
```diff
@@ -70,44 +21,40 @@ object MathUtilities {
...
-        // Pick point such that look ahead point angle to end point is same as angle
-        // of segment, ensuring that we drive toward the correct direction
-        if ((segment.angle - Segment(segment.start, furthestFromStart).angle).abs < tolerance ) {
-          Some(furthestFromStart)
-        } else if ( (segment.angle - Segment(segment.start, closerToStart).angle).abs < tolerance ) {
-          Some(closerToStart)
-        } else {
-          None
-        }
+    val solutions = segment.intersectionWithCircle(center, radius)
```
Comment (PhilipAxelrod) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/unicycle/control/purePursuit/MathUtilities.scala:24]: Rename to clarify it's the intersection with the **line** defined by the segment, not the segment itself. Although we can refactor later

File: core/shared/src/main/scala/com/lynbrookrobotics/potassium/units/SpatialValues.scala [hunk 2]
```diff
@@ -84,28 +84,60 @@ case class Segment(start: Point, end: Point) {
...
-
-    val minY = (start.y min end.y) - Feet(0.01)
-    val maxY = (start.y max end.y) + Feet(0.01)
+  def angle: Angle = {
+    Radians(math.atan2(dy.toFeet, dx.toFeet))
+  }
 
-    toTest.x >= minX && toTest.x <= maxX &&
-      toTest.y >= minY && toTest.y <= maxY
+  def pointClosestToOnLine(pt: Point): Point = {
```
Comment (PhilipAxelrod) [inline core/shared/src/main/scala/com/lynbrookrobotics/potassium/units/SpatialValues.scala:91]: What are you doing?

**Review by PhilipAxelrod — APPROVED** [review]
This is really messy, but given our deadlines, I'm going to merge this and we can refactor later.

File: core/shared/src/main/scala/com/lynbrookrobotics/potassium/units/SpatialValues.scala (hunk 2 above)
Comment (shadaj) [reply core/shared/src/main/scala/com/lynbrookrobotics/potassium/units/SpatialValues.scala:91]: Projecting the point pt onto the line, which results in a new point on the given line that is closest to the pt parameter.

**Pull request merged**